pkg/cosign/verify.go

```diff
@@ -1030,6 +1071,34 @@ func compareSigs(bundleBody string, sig oci.Signature) error {
 	return nil
 }
 
+func comparePublicKey(bundleBody string, sig oci.Signature, co *CheckOpts) error {
+	pemBytes, err := keyBytes(sig, co)
+	if err != nil {
+		return err
+	}
+
+	bundleKey, err := bundleKey(bundleBody)
+	if err != nil {
+		return fmt.Errorf("failed to extract key from bundle: %w", err)
+	}
+
+	decodeSecond, err := base64.StdEncoding.DecodeString(bundleKey)
+	if err != nil {
+		return fmt.Errorf("decoding base64 string %s", bundleKey)
+	}
+
+	// Compare the PEM bytes, to spurious ignore newlines.
+	pemFirst, _ := pem.Decode(pemBytes)
```

Comment:
Should we err if `rest` is not just newlines?

Reply:
Good point: you could stick in some spurious data. It's unexpected and prob doesn't change properties, but let's do it anyway

Reply:
Oops this broke tests... fixing now

Reply:
Fixed